devel/haskell-language-server/Portfile

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           haskell_cabal 1.0
+
+name                haskell-language-server
+version             2.6.0.0
+revision            0
+categories          devel haskell
+maintainers         nomaintainer
+license             GPL-3
+platforms           darwin
+homepage            http://hackage.haskell.org/package/haskell-language-server
+
+description         Official haskell ide support via language server (LSP).
+
+long_description    The haskell-language-server (HLS) project is an \
+                    implementation of a server (a \"language server\") \
+                    for the Language Server Protocol (LSP). A language \
+                    server talks to a client (typically an editor), \
+                    which can ask the server to perform various \
+                    operations, such as reporting errors or providing \
+                    code completions. The advantage of this system is \
+                    that clients and servers can interoperate more \
+                    easily so long as they all speak the LSP \
+                    protocol. In the case of HLS, that means that it \
+                    can be used with many different editors, since \
+                    editor support for the LSP protocol is now \
+                    widespread.
+
+master_sites        https://hackage.haskell.org/package/${name}-${version}
+
+checksums           rmd160  66c0d520c28d2a8e4a1ca0e3665f2aa73ed75582 \
+                    sha256  3655b8d741616f8e32d0181274f80a48d9b8dca47d3fd22675cadfa01ca80e48 \
+                    size    80146
+
+set name_underscore [string map {- _} ${name}]
+set paths_name_underscore   src/Paths_${name_underscore}.hs
+
+# relative paths to ${prefix}
+set hls_libdir      lib/${distname}
+set hls_docdir      share/doc/${distname}/html
+
+
+post-extract {      
+    # Fix for cabal data-files hardcoded path in binary
+    # See:          
+    # https://github.com/commercialhaskell/stack/issues/848
+    # https://github.com/commercialhaskell/stack/issues/4857
+    # https://github.com/haskell/cabal/issues/462
+    # https://github.com/haskell/cabal/issues/3586
+    xinstall -m 0644 -W ${worksrcpath} \
+                    ${filespath}/Paths_NAME.hs ${paths_name_underscore}
+
+    reinplace "s|@PREFIX@|${prefix}|g" \
+                    ${worksrcpath}/${paths_name_underscore}
+    reinplace "s|@NAME@|${name_underscore}|g" \
+                    ${worksrcpath}/${paths_name_underscore}
+    reinplace -E "s|(Version\[\[:space:\]\]+)\\\[\[\[:digit:\]\]+\[\[:space:\]\]*(,\[\[:space:\]\]*\[\[:digit:\]\]+){1,4}\[\[:space:\]\]*\\\]|\\1\[[join [split ${version} .] ,]\]|" \
```

Comment:
this should be a patchfile ; nobody is going to be able to read/understand this `reinplace` a week from now

Reply:
I understand your comment, but in this specific instance the versions are hardcoded as Haskell lists. Because the syntax for Haskell lists will never, ever change, this regex will never, ever change—it's write once and forget about it. Using a patchfile to implement a version stamp would require the unnecessarily onerous regeneration of the patchfile at every version update.

I added a comment to explain the regex.